src/test/java/seedu/address/logic/commands/AddLeaveCommandIntegrationTest.java

```diff
@@ -0,0 +1,57 @@
+package seedu.address.logic.commands;
+
+import static seedu.address.logic.commands.CommandTestUtil.assertCommandFailure;
+import static seedu.address.logic.commands.CommandTestUtil.assertCommandSuccess;
+import static seedu.address.testutil.TypicalIndexes.INDEX_FIRST_LEAVE;
+import static seedu.address.testutil.TypicalLeaves.getTypicalLeavesBook;
+import static seedu.address.testutil.TypicalPersons.getTypicalAddressBook;
+
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import seedu.address.logic.Messages;
+import seedu.address.model.Model;
+import seedu.address.model.ModelManager;
+import seedu.address.model.UserPrefs;
+import seedu.address.model.leave.Leave;
+import seedu.address.model.leave.Range;
+import seedu.address.testutil.LeaveBuilder;
+
+/**
+ * Contains integration tests (interaction with the Model) for {@code AddCommand}.
+ */
+public class AddLeaveCommandIntegrationTest {
```

Comment:
should we write test cases to ensure errors are properly thrown for invalid inputs?

Reply:
just to clarify, this one should not be involving the exceptions thrown in the parser?

Reply:
Ideally we should, since we want to ensure that all exceptions thrown are properly handled to avoid a case where we have a non-responsive app, but in the interest of time, if they are already covered in the parser tests, it's ok to skip them